scripts/install/install_unix.sh

```diff
@@ -13,165 +13,638 @@
+
+banner()
+{
+  printf "\\n"
+  separator
+  printf "| %s\\n" "$*" ;
+  separator
+}
+
+usage()
+{
+  increase_indent
+  USAGE=$(cat <<EOF
+Usage:
+  $(fg_yellow '-v, --version')
+      Defines the version of the observIQ OpenTelemetry Collector.
+      If not provided, this will default to the latest version.
+      Example: '-v 1.2.12' will download 1.2.12.
+
+  $(fg_yellow '-l, --url')
+      Defines the URL that the components will be downloaded from.
+      If not provided, this will default to observIQ OpenTelemetry Collector\'s GitHub releases.
+      Example: '-l http://my.domain.org/observiq-otel-collector' will download from there.
+
+  $(fg_yellow '-x, --proxy')
+      Defines the proxy server to be used for communication by the install script.
+      Example: $(fg_blue -x) $(fg_magenta http\(s\)://server-ip:port/).
+
+  $(fg_yellow '-U, --proxy-user')
+      Defines the proxy user to be used for communication by the install script.
+
+  $(fg_yellow '-P, --proxy-password')
+      Defines the proxy password to be used for communication by the install script.
+
+EOF
+  )
+  info "$USAGE"
+  decrease_indent
+  return 0
+}
+
+force_exit()
+{
+  # Exit regardless of subshell level with no "Terminated" message
+  kill -PIPE $$
+  # Call exit to handle special circumstances (like running script during docker container build)
+  exit 1
+}
+
+error_exit()
+{
+  line_num=$(if [ -n "$1" ]; then command printf ":$1"; fi)
+  error "ERROR ($SCRIPT_NAME$line_num): ${2:-Unknown Error}" >&2
+  if [ -n "$0" ]; then
+    increase_indent
+    error "$*"
+    decrease_indent
+  fi
+  force_exit
+}
+
+print_prereq_line()
+{
+  if [ -n "$2" ]; then
+    command printf "\\n${indent}  - "
+    command printf "[$1]: $2"
+  fi
+}
+
+check_failure()
+{
+  if [ "$indent" != '' ]; then increase_indent; fi
+  command printf "${indent}${fg_red}ERROR: %s check failed!${reset}" "$1"
+
+  print_prereq_line "Issue" "$2"
+  print_prereq_line "Resolution" "$3"
+  print_prereq_line "Help Link" "$4"
+  print_prereq_line "Rerun" "$5"
+
+  command printf "\\n"
+  if [ "$indent" != '' ]; then decrease_indent; fi
+  force_exit
+}
+
+succeeded()
+{
+  increase_indent
+  success "Succeeded!"
+  decrease_indent
+}
+
+failed()
+{
+  error "Failed!"
+}
+
+# This will set all installation variables
+# at the beginning of the script.
+setup_installation()
+{
+    banner "Configuring Installation Variables"
+    increase_indent
+
+    # Installation variables
+    set_os
+    set_os_arch
+    set_package_type
+    set_download_urls
+    set_proxy
+    set_file_name
+
+    success "Configuration complete!"
+    decrease_indent
+}
+
+set_file_name() {
+    package_file_name="${PACKAGE_NAME}_${arch}.${package_type}"
+    out_file_path="$TMP_DIR/$package_file_name"
+}
+
+set_proxy()
+{
+  if [ -n "$proxy" ]; then
+    info "Using proxy from arguments: $proxy"
+    if [ -n "$proxy_user" ]; then
+      while [ -z "$proxy_password" ] && [ ! "$accept_defaults" = "yes" ]; do
```

Comment:
`accept_defaults` is not assigned anywhere, we should remove it from this if clause.